### PR TITLE
✨ feat: Add ability to include tutorial video link for Cosmetics

### PR DESCRIPTION
**Pull request by bryantanjw**
**What's changed**

- Added `videoUrl` column to the Cosmetic and CosmeticShopItem model in schema.
- Updated `CosmeticShopItemUpsertForm` to include a new input field for `videoUrl`.
- Modified `ProfileSidebar` to display a button linking to the `videoUrl ` if available
- Added `videoUrl` to cosmeticShopItemSelect
image

### CODE REVIEW

File: prisma/migrations/20240619181506_add_video_url/migration.sql [hunk 1]
```diff
@@ -0,0 +1 @@
+ALTER TABLE "Cosmetic" ADD COLUMN "videoUrl" TEXT;
```
Comment (lrojas94) [inline prisma/migrations/20240619181506_add_video_url/migration.sql:1]: Why do we want this both in the Cosmetic table and the `CosmeticShopItem` table?

File: src/components/CosmeticShop/cosmetic-shop.util.ts [hunk 2]
```diff
@@ -114,6 +115,15 @@ export const useMutateCosmeticShop = () => {
...
 
+  const upsertCosmeticMutation = trpc.cosmeticShop.upsertCosmetic.useMutation({
+    async onSuccess() {
+      await queryUtils.cosmeticShop.getShopItemsPaged.invalidate();
+    },
+    onError(error) {
+      onError(error, 'Failed to update or create the cosmetic');
+    },
+  });
+
```
Comment (manuelurenah) [inline src/components/CosmeticShop/cosmetic-shop.util.ts:126]: Seems like we are not using this anymore, we can keep it if you think it makes sense

File: src/components/CosmeticShop/cosmetic-shop.util.ts [hunk 3]
```diff
@@ -216,6 +226,10 @@ export const useMutateCosmeticShop = () => {
     },
   });
 
+  const handleUpsertCosmetic = (data: UpsertCosmeticInput) => {
+    return upsertCosmeticMutation.mutateAsync(data);
+  };
+
```
Comment (manuelurenah) [inline src/components/CosmeticShop/cosmetic-shop.util.ts:232]: Ditto

File: src/components/Profile/ProfileSidebar.tsx [hunk 4]
```diff
@@ -273,7 +275,7 @@ export function ProfileSidebar({ username, className }: { username: string; clas
           </Text>
           <Group spacing="xs">
             {(showAllBadges ? badges : badges.slice(0, sizeOpts.badgeCount)).map((award) => {
-              const data = (award.data ?? {}) as { url?: string; animated?: boolean };
+              const data = (award.data ?? {}) as { url?: string; animated?: boolean; videoUrl?: string };
```
Comment (manuelurenah) [inline src/components/Profile/ProfileSidebar.tsx:278]: Just a suggestion
```suggestion
              const data = (award.data ?? {}) as BadgeCosmetic['data'];
```

File: src/components/CosmeticShop/cosmetic-shop.util.ts (hunk 3 above)
Comment (bryantanjw) [reply src/components/CosmeticShop/cosmetic-shop.util.ts:232]: hehe

**Review by manuelurenah — APPROVED** [review]

**Review by manuelurenah — APPROVED** [review]

**Pull request merged**